doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+
+To get configuration values, use the schema object's :ref:`get() <config-schema_object-get>` method.
+The function takes the configuration and the full path to the node as arguments:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_record_hierarchy/http_api.lua
+    :language: lua
+    :start-at: local function apply
+    :end-before: local function stop
+    :dedent:
+
+
+.. _config_utils_schema_transform_configuration:
+
+Transforming configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The module provides methods that transform configuration data based on the schema,
+for example, :ref:`apply_default() <config-schema_object-apply_default>`,
+:ref:`merge() <config-schema_object-merge>`, :ref:`set() <config-schema_object-set>`.
+
+The following sample shows how to apply default values from the schema to fill
+missing configuration fields:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_annotations/http_api.lua
+    :language: lua
+    :start-at: local function apply
+    :end-before: local function stop
+    :dedent:
+
+
+.. _config_utils_schema_env-vars:
+
+Parsing environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :ref:`schema.fromenv() <config-utils-schema-fromenv>` function allows getting
+configuration values from environment variables. The example below shows how to do
+this by adding a user-defined annotation ``env``:
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_fromenv/http_api.lua
+    :language: lua
+    :start-at: local listen_address_schema
+    :end-before: local function validate
+    :dedent:
+
+The function also uses schema object methods:
+
+-   :ref:`pairs() <config-schema_object-pairs>` to iterate over the schema nodes.
+-   :ref:`set() <config-schema_object-set>` to assign configuration values.
+
+
+.. _api-reference-config-utils-schema:
+
+API Reference
+-------------
+
+..  container:: table
+
+    ..  rst-class:: left-align-column-1
+    ..  rst-class:: left-align-column-2
+
+    ..  list-table::
+        :widths: 45 55
+
+        *   -   **Functions**
+            -
+
+        *   -   :ref:`schema.array() <config-utils-schema-array>`
+            -   Create an array
+
+        *   -   :ref:`schema.enum() <config-utils-schema-enum>`
+            -   Create an enum
+
+        *   -   :ref:`schema.fromenv() <config-utils-schema-fromenv>`
+            -   Parse a value from an environment variable
+
+        *   -   :ref:`schema.map() <config-utils-schema-map>`
+            -   Create a map
+
+        *   -   :ref:`schema.new() <config-utils-schema-new>`
+            -   Create a schema
+
+        *   -   :ref:`schema.record() <config-utils-schema-record>`
+            -   Create a record
+
+        *   -   :ref:`schema.scalar() <config-utils-schema-scalar>`
+            -   Create a scalar
+
+        *   -   :ref:`schema.set() <config-utils-schema-set>`
+            -   Create a set
```

Comment:
TBH, I don't like that all the functions are in one bag, while we have separate groups in the head. At least, I would separate schema node constructors into its own group.

Reply:
This list is pretty short, and items have obvious descriptions (`Create a .. [node]`), so I'd keep the structure plain.
Also, I see only one function that is not a constructor - `fromenv`. It would look rather weird to create a dedicated group for it. Perhaps, we can move it to the bottom of the list.

Reply:
I see schema node constructors (scalar, record, map, array), constructor shortcuts (enum, set), schema object constructor (new) and a separate functions (fromenv).

It would be schema node constructors (scalar, record, map, array, enum, set) and other functions (new, fromenv).

I'll resolve this comment -- it is a thing of taste.